### PR TITLE
Feat: mobile uni link support (android + ios)

### PR DESCRIPTION
**Pull request by beelchester**
https://github.com/rustdesk/rustdesk/discussions/4669
Connect to ID using uni/deep link on mobile

Format: `rustdesk://[id]` or `rustdesk://connection/new/[id]` or `rustdesk://connect/[id]` or `rustdesk://play/[id]`
Ex: `rustdesk://13423325234` or `rustdesk://connection/new/13423325234`...

**File transfer:**
Format: `rustdesk://file-transfer/[id]`
Ex: `rustdesk://file-transfer/13423325234`

Works on cold boot as well as when the app is running in the background.

https://github.com/rustdesk/rustdesk/assets/73148455/acc800f1-8e6a-44ee-8f90-937ba6991b6f

https://github.com/rustdesk/rustdesk/assets/73148455/206397fc-e1fb-46df-a19b-128572ae8968

https://github.com/rustdesk/rustdesk/assets/73148455/39148305-58e1-48f0-93f4-ce449bdba0ab

### CODE REVIEW

File: flutter/lib/common.dart [hunk 1]
```diff
@@ -1823,10 +1827,15 @@ StreamSubscription? listenUniLinks({handleByFlutter = true}) {
...
-        bind.sendUrlScheme(url: uri.toString());
+      if (!isMobile){
+        if (handleByFlutter) {
+          handleUriLink(uri: uri);
+        } else {
+          bind.sendUrlScheme(url: uri.toString());
+        }
+      }
+      else {
+        handleUriLinkMobile(uri.toString());
```
Comment (rustdesk) [inline flutter/lib/common.dart]: format?

File: flutter/lib/common.dart [hunk 2]
```diff
@@ -1823,10 +1827,15 @@ StreamSubscription? listenUniLinks({handleByFlutter = true}) {
   final sub = uriLinkStream.listen((Uri? uri) {
     debugPrint("A uri was received: $uri. handleByFlutter $handleByFlutter");
     if (uri != null) {
-      if (handleByFlutter) {
-        handleUriLink(uri: uri);
-      } else {
-        bind.sendUrlScheme(url: uri.toString());
+      if (!isMobile){
```
Comment (rustdesk) [inline flutter/lib/common.dart]: format?

File: flutter/lib/common.dart (hunk 1 above)
Comment (rustdesk) [reply flutter/lib/common.dart]: Please use auto format

File: flutter/lib/common.dart [hunk 3]
```diff
@@ -1844,6 +1853,14 @@ enum UriLinkType {
...
 }
 
+void handleUriLinkMobile(String uri) {
+  var context = Get.context;
+  var uri_id = uri.split("//").last;
+  if (context != null && uri_id.isNotEmpty){
+    connect(context, uri_id);
+  }
+}
+
```
Comment (rustdesk) [inline flutter/lib/common.dart]: Please use handleUriLink, let mobile and desktop support the same uri format.

Comment (sahilyeole) [reply flutter/lib/common.dart:1863]: handleUriLink has multiple cases apart from connection (such as file_transfer, relay, etc). Should I support those as well? or just for connection?

**Comment by sahilyeole** [comment]
>Please use handleUriLink, let mobile and desktop support the same uri format.

Done

File: flutter/lib/common.dart (hunk 3 above)
Comment (rustdesk) [reply flutter/lib/common.dart]: For all that are applicable to mobile.

File: flutter/lib/common.dart [hunk 4]
```diff
@@ -1963,7 +1963,7 @@ List<String>? urlLinkToCmdArgs(Uri uri) {
     command = '--connect';
     id = uri.path.substring("/new/".length);
   } else if (['connect', "play", 'file-transfer', 'port-forward', 'rdp']
-      .contains(uri.authority)) {
+      .contains(uri.authority) && !isMobile) {
     command = '--${uri.authority}';
```
Comment (rustdesk) [inline flutter/lib/common.dart:1968]: 'connect', "play", 'file-transfer' should work for mobile also.

**Comment by xlionjuan** [comment]
I tested on Windows:

I use the "shortcut" option on Desktop and I founded out adding `/` at the end of the URL will cause `ID not exist error` with `file-transfer` URL, for example:




|   | |
|----------|-------------|
|`rustdesk://1234567890`|      OK      | 
|`rustdesk://1234567890/`|      OK      | 
|`rustdesk://1234567890/r`|      OK      | 
|   | |
|`rustdesk://file-transfer/1234567890`|      OK      | 
|`rustdesk://file-transfer/1234567890/`|     *ID not exist error*      | 
|`rustdesk://file-transfer/1234567890/r`|      OK      |

**Pull request merged**